src/SkeletalTrapezoidation.cpp

```diff
@@ -0,0 +1,3083 @@
+        coord_t end_x = projected_x(end);
+
+        float bound = 0.5 / tan((M_PI - transitioning_angle) * 0.5);
+        coord_t marking_start_x = - d * bound;
+        coord_t marking_end_x = d * bound;
+        Point marking_start = middle + x_axis_dir * marking_start_x / x_axis_length;
+        Point marking_end = middle + x_axis_dir * marking_end_x / x_axis_length;
+        coord_t dir = 1;
+        if (start_x > end_x)
+        {
+            dir = -1;
+            std::swap(marking_start, marking_end);
+            std::swap(marking_start_x, marking_end_x);
+        }
+
+        Point a = start;
+        Point b = end;
+        std::vector<Point> ret;
+        ret.emplace_back(a);
+        
+        bool add_marking_start = marking_start_x * dir > start_x * dir;
+        bool add_marking_end = marking_end_x * dir > start_x * dir;
+        
+        Point ab = b - a;
+        coord_t ab_size = vSize(ab);
+        coord_t step_count = (ab_size + discretization_step_size / 2) / discretization_step_size;
+        if (step_count % 2 == 1)
+        {
+            step_count++; // enforce a discretization point being added in the middle
+        }
+        for (coord_t step = 1; step < step_count; step++)
+        {
+            Point here = a + ab * step / step_count;
+            coord_t x_here = projected_x(here);
+            if (add_marking_start && marking_start_x * dir < x_here * dir)
+            {
+                ret.emplace_back(marking_start);
+                add_marking_start = false;
+            }
+            if (add_marking_end && marking_end_x * dir < x_here * dir)
+            {
+                ret.emplace_back(marking_end);
+                add_marking_end = false;
+            }
+            ret.emplace_back(here);
+        }
+        if (add_marking_end && marking_end_x * dir < end_x * dir)
+        {
+            ret.emplace_back(marking_end);
+        }
+        ret.emplace_back(b);
+        return ret;
+    }
+}
+
+
+bool SkeletalTrapezoidation::computePointCellRange(vd_t::cell_type& cell, Point& start_source_point, Point& end_source_point, vd_t::edge_type*& starting_vd_edge, vd_t::edge_type*& ending_vd_edge, const std::vector<Point>& points, const std::vector<Segment>& segments)
+{
+    if (cell.incident_edge()->is_infinite())
+    {
+        return false;
+    }
+    // check if any point of the cell is inside or outside polygon
+    // copy whole cell into graph or not at all
+    
+    const Point source_point = VoronoiUtils::getSourcePoint(cell, points, segments);
+    const PolygonsPointIndex source_point_index = VoronoiUtils::getSourcePointIndex(cell, points, segments);
+    Point some_point = VoronoiUtils::p(cell.incident_edge()->vertex0());
+    if (some_point == source_point)
+    {
+        some_point = VoronoiUtils::p(cell.incident_edge()->vertex1());
+    }
+    if (!LinearAlg2D::isInsideCorner(source_point_index.prev().p(), source_point_index.p(), source_point_index.next().p(), some_point))
```

Comment:
Why is a cell outside of the polygon in this case?

Reply:
I figured it out and documented it here (after I've pushed the commits, which I will do promptly.).